generators/server/templates/src/main/java/package/service/dto/UserDTO.java.ejs

```diff
@@ -110,11 +113,11 @@ public class <%= asDto('User') %> {
     }
 
     <%_ } _%>
-    public <% if (databaseType === 'mongodb' || databaseType === 'neo4j' || databaseType === 'couchbase' || databaseType === 'cassandra' || authenticationType === 'oauth2') { %>String<% } else { %>Long<% } %> getId() {
+    public <%= userPk %> getId() {
```

Comment:
```suggestion
    public <%= user.primaryKeyType %> getId() {
```

Reply:
Nice, I haven't noticed that there is a user.primaryKeyType variable accessible!